dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java

```diff
@@ -454,9 +454,9 @@ private SqlBinds getSqlAndSqlParamsMap(String sql) {
         String rgexo = "['\"]*\\!\\{(.*?)\\}['\"]*";
         sql = replaceOriginalValue(sql, rgexo, paramsMap);
         // replace the ${} of the SQL statement with the Placeholder
-        String formatSql = sql.replaceAll(rgex, "?");
+        // String formatSql = sql.replaceAll(rgex, "?");
         // Convert the list parameter
-        formatSql = ParameterUtils.expandListParameter(sqlParamsMap, formatSql);
+        String formatSql = ParameterUtils.expandListParameter(sqlParamsMap, sql, rgex);
```

Comment:
The regex shouldn't exposed to outside.

Reply:
I have reused `AbstractTask.rgex` here. Should I reference `AbstractTask.rgex` in `ParameterUtils` or should I extract this rgex into a constant class? I apologize for not being familiar with this code specification.

Reply:
I extract this rgex into `TaskConstants`